Fix(Curriculum): Constructor function not explained before ES6 #17865

hilliarj opened the pull request:
<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #17865

Commit:
Fix(Curriculum): Constructor function not explained before ES6 #17865

thecodingaviator commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md:
```diff
@@ -13,8 +13,9 @@ In ES5, we usually define a constructor function, and use the <code>new</code> k
-UpperCamelCase should be used by convention for ES6 class names, as in <code>SpaceShuttle</code> used above.
+<strong>Notes</strong><br>
+1. UpperCamelCase should be used by convention for ES6 class names, as in <code>SpaceShuttle</code> used above.
+2. The constructor method is a special method for creating and initializing an object created with a class. Learn more about it in the Object Oriented Programming section of the Javascript Algorithms And Data Structures Certification
```
```suggestion
2. The constructor method is a special method for creating and initializing an object created with a class. You will learn more about it in the Object Oriented Programming section of the Javascript Algorithms And Data Structures Certification
```
This is a better wording since if anyone is going through the curriculum, they won't have to hop on to another challenge from in between somewhere else

thecodingaviator reviewed (changes requested):
Have a look at the comment

thecodingaviator committed:
Update curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md

Co-Authored-By: hilliarj <hilliarj@dickinson.edu>

hilliarj commented:
Updated the PR. Thanks for the suggestion!

thecodingaviator reviewed (approved):
LGTM! 👍

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md:
```diff
@@ -13,8 +13,9 @@ In ES5, we usually define a constructor function, and use the <code>new</code> k
 The class syntax simply replaces the constructor function creation:
 <blockquote>class SpaceShuttle {<br>&nbsp;&nbsp;constructor(targetPlanet){<br>&nbsp;&nbsp;&nbsp;&nbsp;this.targetPlanet = targetPlanet;<br>&nbsp;&nbsp;}<br>}<br>const zeus = new SpaceShuttle('Jupiter');</blockquote>
 Notice that the <code>class</code> keyword declares a new function, and a constructor was added, which would be invoked when <code>new</code> is called - to create a new object.<br>
-<strong>Note</strong><br>
```
Per discussions on issue https://github.com/freeCodeCamp/freeCodeCamp/issues/35464, this should be:
```suggestion
<strong>Notes:</strong><br>
```

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md:
```diff
@@ -13,8 +13,9 @@ In ES5, we usually define a constructor function, and use the <code>new</code> k
-<strong>Note</strong><br>
-UpperCamelCase should be used by convention for ES6 class names, as in <code>SpaceShuttle</code> used above.
+<strong>Notes</strong><br>
+1. UpperCamelCase should be used by convention for ES6 class names, as in <code>SpaceShuttle</code> used above.
```
Per same discussion on issue, this should be a unordered list item:
```suggestion
* UpperCamelCase should be used by convention for ES6 class names, as in <code>SpaceShuttle</code> used above.
```

RandellDawson replied:
You can not use GitHub markdown for the Description or Instruction sections.  You must use an unordered list like:
```
<ul>
  <li>...</li>
  <li>..</li>
</ul>
```
Make sure you are testing all of your changes locally so you can see how it will actually render on the site.

RandellDawson commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function.english.md:
```diff
@@ -13,8 +13,9 @@ In ES5, we usually define a constructor function, and use the <code>new</code> k
-UpperCamelCase should be used by convention for ES6 class names, as in <code>SpaceShuttle</code> used above.
+<strong>Notes</strong><br>
+1. UpperCamelCase should be used by convention for ES6 class names, as in <code>SpaceShuttle</code> used above.
+2. The constructor method is a special method for creating and initializing an object created with a class. You will learn more about it in the Object Oriented Programming section of the Javascript Algorithms And Data Structures Certification
```
Per same discussion on issue, this should be a unordered list item:
```suggestion
* The constructor method is a special method for creating and initializing an object created with a class. You will learn more about it in the Object Oriented Programming section of the Javascript Algorithms And Data Structures Certification
```

RandellDawson replied:
Same issue as pointed out above.

RandellDawson reviewed (changes requested):
See my suggestions.

RandellDawson reviewed (changes requested):
You can not use GitHub markdown.  Please test any non-content changes like HTML, CSS, or JavaScript locally first, before creating PR.

Commit:
Update to #17865 PR

Fix(Curriculum): Constructor function not explained before ES6. Used an unordered list and the syntax that others recommended.

hilliarj commented:
Sorry for the delay, been busy with tests at college and family affairs.

RandellDawson commented:
LGTM.  Just waiting on #34538 to be reviewed/merged before merging this one.

cmccormack reviewed (approved):
This looks good to me 👍.  Thank you for your contribution!

RandellDawson commented:
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉
